wakuv2/waku.go

```diff
@@ -970,23 +979,78 @@ func (w *Waku) broadcast() {
 	}
 }
 
+func (w *Waku) checkIfMessagesStored() {
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-w.ctx.Done():
+			w.logger.Debug("Stop the look for message stored check")
+			return
+		case <-ticker.C:
+			w.logger.Debug("Running loop for messages stored check")
+			w.sendMsgIDsMu.Lock()
+			pubsubTopics := make([]string, 0, len(w.sendMsgIDs))
+			pubsubMessageIds := make([][]gethcommon.Hash, 0, len(w.sendMsgIDs))
+			for pubsubTopic, subMsgs := range w.sendMsgIDs {
+				var queryMsgIds []gethcommon.Hash
+				for msgID, sendTime := range subMsgs {
```

Comment:
just to make sure, here you are taking max 20 random messages from the one sent to check on store? maybe we should use a sorted map (or any sorted struct)